tests/next_tests/unit_tests/program_processor_tests/runners_tests/test_dace.py

```diff
@@ -0,0 +1,121 @@
+        pytest.skip("DaCe-specific testcase.")
+
+    @gtx.field_operator
+    def testee(
+        a: cases.IField,
+        a_idx: cases.IField,
+        unused_field: cases.IField,
+        a0: int32,
+        a1: int32,
+        a2: int32,
+        unused_scalar: int32,
+    ) -> cases.IField:
+        t0 = where(a_idx == 0, a + a0, a)
+        t1 = where(a_idx == 1, t0 + a1, t0)
+        t2 = where(a_idx == 2, t1 + a2, t1)
+        return t2
+
+    numpy_ref = lambda a, a0, a1, a2: [a[0] + a0, a[1] + a1, a[2] + a2, *a[3:]]
+
+    a = cases.allocate(cartesian_case, testee, "a")()
+    a_index = cases.allocate(cartesian_case, testee, "a_idx", strategy=cases.IndexInitializer())()
+    a_offset = np.random.randint(1, 100, size=4, dtype=np.int32)
+    unused_field = cases.allocate(cartesian_case, testee, "unused_field")()
+    out = cases.allocate(cartesian_case, testee, cases.RETURN)()
+
+    # Wrap `compiled_sdfg.CompiledSDFG.fast_call` with mock object
+    mock_fast_call = unittest.mock.MagicMock()
+    mock_fast_call_attr = getattr(compiled_sdfg.CompiledSDFG, "fast_call")
+
+    def mocked_fast_call(self, *args, **kwargs):
+        mock_fast_call.__call__(*args, **kwargs)
+        return mock_fast_call_attr(self, *args, **kwargs)
+
+    monkeypatch.setattr(compiled_sdfg.CompiledSDFG, "fast_call", mocked_fast_call)
+
+    # Wrap `compiled_sdfg.CompiledSDFG._construct_args` with mock object
+    mock_construct_args = unittest.mock.MagicMock()
+    mock_construct_args_attr = getattr(compiled_sdfg.CompiledSDFG, "_construct_args")
+
+    def mocked_construct_args(self, *args, **kwargs):
+        mock_construct_args.__call__(*args, **kwargs)
+        return mock_construct_args_attr(self, *args, **kwargs)
+
+    monkeypatch.setattr(compiled_sdfg.CompiledSDFG, "_construct_args", mocked_construct_args)
+
+    # Reset mock objects and run/verify GT4Py program
+    def verify_testee():
+        mock_construct_args.reset_mock()
+        mock_fast_call.reset_mock()
+        cases.verify(
+            cartesian_case,
+            testee,
+            a,
+            a_index,
+            unused_field,
+            *a_offset,
+            out=out,
+            ref=numpy_ref(a.asnumpy(), *a_offset[0:3]),
+        )
+        mock_fast_call.assert_called_once()
+
+    # On first run, the SDFG arguments will have to be constructed
+    verify_testee()
+    mock_construct_args.assert_called_once()
+    fast_call_args = mock_fast_call.call_args.args
+
+    # Now modify the scalar arguments, used and unused ones: reuse previous SDFG arguments
+    for i in range(4):
+        a_offset[i] += 1
+        verify_testee()
+        mock_construct_args.assert_not_called()
+        assert mock_fast_call.call_args.args == fast_call_args
+
+    # Modify content of current buffer: reuse previous SDFG arguments
+    a[0] += 1
+    verify_testee()
+    mock_construct_args.assert_not_called()
+    assert mock_fast_call.call_args.args == fast_call_args
```

Comment:
Here I agree that the arguments (the address to the memory storing the argument) do not change as you modify the content, which is different than above.